app/services/foreman_templates/template_importer.rb

```diff
@@ -0,0 +1,154 @@
+
+        unless metadata['kind']
+          puts '  Error: Must specify template kind'
+          next
+        end
+
+        begin
+          # Expects a return of { :diff, :status, :result }
+          data = if metadata['model'].present?
+                   metadata['model'].constantize.import!(name, text, metadata)
+                 else
+                   # For backwards-compat before "model" metadata was added
+                   case metadata['kind']
+                   when 'ptable'
+                     Ptable.import!(name, text, metadata)
+                   when 'job_template'
+                     # TODO: update REX templates to have `model` and delete this
+                     update_job_template(name, text)
+                   else
+                     ProvisioningTemplate.import!(name, text, metadata)
+                   end
+                 end
+
+          if data[:status] == true && @verbose
+            puts data[:result]
+            puts data[:diff]
+          elsif data[:status] == false
+            puts "Template \"#{name}\": #{data[:result]}"
+          end
+        rescue NoKindError
+          puts "  Error: Unknown template type '#{metadata['kind']}'"
+          next
+        end
+      end
+    end
+
+    def get_default_branch(repo)
+      branch_names = repo.branches.map(&:name).uniq
+
+      # Always use develop on Foreman-nightly, if present, or else relevant stable branch
+      target = SETTINGS[:version].tag == 'develop' ? 'develop' : "#{SETTINGS[:version].short}-stable"
+      return target if branch_names.include?(target)
+
+      # stay on default branch as fallback
+      nil
+    end
+
+    def parse_metadata(text)
+      # Pull out the first erb comment only - /m is for a multiline regex
+      extracted = text.match(/<%\#(.+?).-?%>/m)
+      extracted.nil? ? {} : YAML.load(extracted[1])
+    end
+
+    def update_job_template(name, text)
+      return {
+        :status => false,
+        :result => 'Skipping job template import, remote execution plugin is not installed.'
+      } unless defined?(JobTemplate)
+      template = JobTemplate.import(
+        text.sub(/^name: .*$/, "name: #{name}"),
+        :update => true
+      )
+
+      c_or_u = template.new_record? ? 'Created' : 'Updated'
+      id_string = ('id' + template.id) rescue ''
+
+      if template.template != template.template_was
+        diff = Diffy::Diff.new(
+          template.template_was,
+          template.template,
+          :include_diff_info => true
+        ).to_s(:color)
+      end
+
+      result = "  #{c_or_u} Template #{id_string}:#{name}"
+      { :diff => diff, :status => template.save, :result => result }
+    end
+
+    def purge!
+#      @kind    = args[:kind] || 'something'
+      @match   = args[:negate] ? 'NOT LIKE ' : 'LIKE '
+      @prefix  = args[:prefix] || 'Community '
+      @filter  = args[:filter] || "name #{match} '#{prefix}%'"
+      @verbose = args[:verbose] || false
```

Comment:
you don't need to use `args` here - that only available in `initialize` - but `@` variables are class-wide so you can just use them right away, so long as they get set properly in `initialize`. Currently, `@match` is the only one not set up, so if you drop that, you don't need any changes in `initialize` and can just use `@prefix/filter/verbose` directly (as `import!` and `parse_files!` do)